### ISSUE TITLE
Can't change settings page

### ISSUE THREAD

**Issue by nyalldawson** [opengisch/QField#1545]
On master it's impossible to change the settings page -- while the header tab updates, the visible settings don't change accordingly.

### Reproduction Steps:

1. [First Step]
2. [Second Step]
3. [Other Steps...]

### Expected behavior:

[Describe expected behavior here]

### Observed behavior:

[Describe observed behavior here]

### Screenshots and GIFs

![Screenshots and GIFs which follow reproduction steps to demonstrate the problem](url)

[Please also attach additional files if a specific project/dataset is useful to investigate the problem.]

**QField version:** [Enter QField version here including build number found in the about screen after the release name example 1.2.0 (10200993)]

## Additional information:

* Problem started happening recently, didn't happen in an older version of QField: [Yes/No]
* Problem can be reliably reproduced, doesn't happen randomly: [Yes/No]
* Problem happens with all files and projects, not only some files or projects: [Yes/No]

**Comment by nyalldawson** [comment]
Hmmmm - changing projects fixed this 🤔

**Comment by suricactus** [comment]
If you click somewhere in the contents of the tabs, clicking on the tab header no longer switches the tab. Bug confirmed. Currently looking at this.

**Comment by suricactus** [comment]
@nirvn I found it, it is the new zoom being too greedy:
https://github.com/opengisch/QField/commit/2128d3d68d4207d4924e44073339f1b5956a2947#diff-4fa2cff8dee0271b57c896b75f4e9934c7667f7d1983905655d23a8ec41eb7a6L155-L158

I have played with different combinations of `grabPermissions`, seems that `CanTakeOverFromItems` is stealing the drag from everything else (e.g. scrolling when the legend is too long, etc). The easiest solution would be to disable the `DragHandler` whenever the map is overlayed. Feel free to suggest better solution.
